### PR TITLE
Allow null User fields

### PR DESCRIPTION
**Pull request by liviu-timar**
### 🎯 Goal

Made some of `User` fields nullable and `null` by default.

In `CoordinatorSocket.authenticate()` we pass:
- `null` if we don't want the field's value to change
- `""` or `emptyMap()` or `emptyList()` if we want to reset the field
- a `value` if we want to set the field
- `null` if the field is blank (whitespace)

### 🧪 Testing

❗ Needs testing

### CODE REVIEW

File: demo-app/src/main/kotlin/io/getstream/video/android/ui/join/CallJoinScreen.kt [hunk 1]
```diff
@@ -234,7 +234,7 @@ private fun CallJoinHeader(
         Text(
             modifier = Modifier.weight(1f),
             color = Color.White,
-            text = user?.name?.ifBlank { user?.id }?.ifBlank { user!!.custom["email"] }.orEmpty(),
+            text = user?.userNameOrId ?: "",
```
Comment (skydoves) [inline demo-app/src/main/kotlin/io/getstream/video/android/ui/join/CallJoinScreen.kt]: We can make this `user?.userNameOrId.orEmpty()`

File: demo-app/src/main/kotlin/io/getstream/video/android/util/StreamVideoInitHelper.kt [hunk 2]
```diff
@@ -168,8 +168,8 @@ object StreamVideoInitHelper {
 
         val chatUser = io.getstream.chat.android.models.User(
             id = user.id,
-            name = user.name,
-            image = user.image,
+            name = user.name ?: "",
+            image = user.image ?: "",
```
Comment (skydoves) [inline demo-app/src/main/kotlin/io/getstream/video/android/util/StreamVideoInitHelper.kt]: We can also utilize `orEmpty()` here as well.

**Review by skydoves — APPROVED** [review]
Looks good to me! I left trivial comments for syntax.

File: demo-app/src/main/kotlin/io/getstream/video/android/ui/join/CallJoinScreen.kt (hunk 1 above)
Comment (liviu-timar) [reply demo-app/src/main/kotlin/io/getstream/video/android/ui/join/CallJoinScreen.kt]: Thanks for noticing. Used it in another place, but missed these ones. 👍

**Review by aleksandar-apostolov — APPROVED** [review]

**Pull request merged**